<launch>
  <rosparam command="load" file="$(find dasom_controllers)/config/link.yaml"/>
<rosparam command="load" file="$(find dasom_controllers)/config/admittance.yaml"/>

<!--For DoB&Current based position control-->
  <arg name="robot_name"      default="dasom"/>

  <arg name="position_p_gain"          default="700"/>
  <arg name="position_i_gain"          default="0"/>
  <arg name="position_d_gain"          default="400"/>

  <arg name="dob_p_gain"              default="0"/>
  <arg name="dob_i_gain"              default="0"/>
  <arg name="dob_d_gain"              default="0"/>
  <arg name="dob_polar_moment"        default="0"/>

  <arg name="polar_moment_1"             default="0.0027"/>
  <arg name="polar_moment_2"             default="0.0027"/>
  <arg name="Cut_Off_Freq"             default="5"/>

  <arg name="cut_off_freq_current"       default="5"/>

<!--end-->

  <param name="robot_name"    value="$(arg robot_name)"/>

  <param name="position_p_gain"        value="$(arg position_p_gain)"/>
  <param name="position_i_gain"        value="$(arg position_i_gain)"/>
  <param name="position_d_gain"        value="$(arg position_d_gain)"/>

  <param name="polar_moment_1"        value="$(arg polar_moment_1)"/>
  <param name="polar_moment_2"        value="$(arg polar_moment_2)"/>
  <param name="Cut_Off_Freq"           value="$(arg Cut_Off_Freq)"/>


  <param name="cut_off_freq_current"           value="$(arg cut_off_freq_current)"/>


  <node name="dasom_manipulator_control" pkg="dasom_controllers" type="dasom_manipulator_control" required="true" output="screen"> 
    <param name="dob_p_gain"              value="$(arg dob_p_gain)"/>
    <param name="dob_i_gain"              value="$(arg dob_i_gain)"/>
    <param name="dob_d_gain"              value="$(arg dob_d_gain)"/>
    <param name="dob_polar_moment"        value="$(arg dob_polar_moment)"/>
  </node>

</launch>
